test/AST-Quake/vector-ctor-initlist-int.cpp

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// Test for std::vector<int> initializer_list constructor support
+
+// RUN: cudaq-quake %s | cudaq-opt | FileCheck %s
+
+#include "cudaq.h"
+
+__qpu__ void testInt() {
+  cudaq::qreg<3> q;
+  std::vector<int> index{0, 1, 2};
+  ry(M_PI_2, q[index[0]]);
+  ry(M_PI_2, q[index[1]]);
+  ry(M_PI_2, q[index[2]]);
+}
+
+
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__function_testInt._Z7testIntv() attributes {"cudaq-entrypoint", "cudaq-kernel", no_this} {
+// CHECK:           %[[VAL_0:.*]] = arith.constant 1.5707963267948966 : f64
+// CHECK:           %[[VAL_1:.*]] = arith.constant 2 : i32
+// CHECK:           %[[VAL_2:.*]] = arith.constant 1 : i32
+// CHECK:           %[[VAL_3:.*]] = arith.constant 0 : i32
```

Comment:
Might want to make these CHECK-DAG lines since the order of the constants can possibly change on different systems.